README.md

```diff
@@ -1 +1,155 @@
+- **Efficient Search**: Quickly find books using category-based search functionality.
+
+
+## Technologies Used
+
+
+- **Java**: The backbone of the project, providing robust and versatile programming capabilities.
+- **Spring Boot**: Facilitates the creation of production-ready, standalone Java applications.
+- **Spring Security**: Ensures secure authentication and authorisation mechanisms.
+- **JWT (JSON Web Token)**: Enables secure information transfer between parties as a JSON object.
+- **Spring Data JPA**: Simplifies database interactions by reducing boilerplate code.
+- **MapStruct**: Automatically maps between Data Transfer Objects (DTOs) and entities.
+- **Lombok**: Minimizes boilerplate code, improving readability and maintainability.
+- **Maven**: Manages project dependencies and streamlines the build process.
+- **Liquibase**: Controls database changes, ensuring consistency across environments.
+- **Jackson**: Handles efficient JSON parsing and generation.
+- **Swagger**: Generates interactive API documentation.
+- **MySQL**: Manages and stores application data as a relational database system.
+- **Docker**: Develops, ships, and runs applications within containers for consistent environments.
+
+
+## Layered Architecture
+
+
+- **Controller Layer**: Manages incoming HTTP requests and client communications.
+- **Service Layer**: Contains business logic, processing data, and interacting with repositories.
+- **Repository Layer**: Handles direct database interactions for querying and managing data.
+- **Security Layer**: Manages authentication and authorisation processes.
+- **DTO Layer**: Transfers data between application layers, including only necessary information.
+- **Mapper Layer**: Converts between object models to ensure data consistency.
+- **Test Layer**: Validates functionality through unit and integration tests.
+
+
+## Database Entities
+
+
+- **Book**: Represents a book with metadata such as title, author, and publication year.
+- **Category**: Classifies books into genres or subjects.
+- **Shopping Cart**: Temporarily holds selected products for potential purchase.
+- **Cart Item**: Specifies a book and its quantity in the shopping cart.
+- **User**: Registered individual with personal details and assigned permissions.
+- **Role**: Defines user permissions, such as USER or ADMIN.
+- **Order**: Records completed transactions, capturing purchase details.
+- **Order Item**: Details specific books purchased in an order.
+
+
+## Access to Endpoints
+
+
+### User
+- 🟩 **POST** `/auth/registration`: User registration.
+- 🟩 **POST** `/auth/login`: User login.
+
+
+### Book
+- 🟨 **GET** `/books`: Retrieve book catalogue.
+- 🟨 **GET** `/books/{id}`: View book details.
+- 🟨 **GET** `/books/search`: Search books.
+- 🟥 **POST** `/books`: Add a new book.
+- 🟥 **PUT** `/books/{id}`: Update book details.
+- 🟥 **DELETE** `/books/{id}`: Delete book.
```

Comment:
I do not understand why some requests are red and some are yellow, so please add a legend(explanation) of the color-coding at the top of that topic